Kickstarter-iOS/Views/Controllers/CreatePasswordTableViewController.swift

```diff
@@ -0,0 +1,282 @@
+import Library
+import Prelude
+import Result
+import ReactiveSwift
+import UIKit
+
+enum CreatePasswordRow: CaseIterable {
+  case newPassword
+  case confirmNewPassword
+}
+
+extension CreatePasswordRow {
+  var placeholder: String {
+    return Strings.signup_input_fields_password()
+  }
+
+  var returnKeyType: UIReturnKeyType {
+    switch self {
+    case .newPassword: return .next
+    case .confirmNewPassword: return .done
+    }
+  }
+
+  var title: String {
+    switch self {
+    case .newPassword: return Strings.New_password()
+    case .confirmNewPassword: return Strings.Confirm_password()
+    }
+  }
+}
+
+@objc protocol CreatePasswordTableViewControllerType {
+  @objc optional func newPasswordTextFieldChanged(_ sender: UITextField)
+  @objc optional func newPasswordTextFieldDidReturn(_ sender: UITextField)
+  @objc optional func newPasswordConfirmationTextFieldChanged(_ sender: UITextField)
+  @objc optional func newPasswordConfirmationTextFieldDidReturn(_ sender: UITextField)
+}
+
+final class CreatePasswordTableViewController: UITableViewController {
+  // MARK: - Properties
+
+  private let viewModel: CreatePasswordViewModelType = CreatePasswordViewModel()
+  weak var delegate: CreatePasswordViewControllerDelegate?
```

Comment:
I wonder if it might make more sense for the delegate type to be `CreatePasswordTableViewControllerDelegate` since it's the class of type `CreatePasswordTableViewController` that is *delegating* work to another object 🤔

Reply:
![right](https://media.giphy.com/media/O3TGxo7yvCmic/giphy.gif)